src/OpenTelemetry.Instrumentation.Dependencies/Implementation/SqlEventSourceListener.netfx.cs

```diff
@@ -0,0 +1,131 @@
+﻿// <copyright file="SqlEventSourceListener.netfx.cs" company="OpenTelemetry Authors">
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// </copyright>
+#if NETFRAMEWORK
+using System;
+using System.Data.SqlClient;
+using System.Diagnostics;
+using System.Diagnostics.Tracing;
+using System.Reflection;
+using OpenTelemetry.Trace;
+
+namespace OpenTelemetry.Instrumentation.Dependencies.Implementation
+{
+    /// <summary>
+    /// .NET Framework SqlClient doesn't emit DiagnosticSource events.
+    /// We hook into its EventSource if it is available:
+    /// See: <a href="https://github.com/microsoft/referencesource/blob/3b1eaf5203992df69de44c783a3eda37d3d4cd10/System.Data/System/Data/Common/SqlEventSource.cs#L29">reference source</a>.
+    /// </summary>
+    internal class SqlEventSourceListener : EventListener
+    {
+        internal const string ActivitySourceName = "SqlClient";
+        internal const string ActivityName = ActivitySourceName + ".Execute";
+
+        private static readonly Version Version = typeof(SqlEventSourceListener).Assembly.GetName().Version;
+        private static readonly ActivitySource SqlClientActivitySource = new ActivitySource(ActivitySourceName, Version.ToString());
+        private static readonly EventSource SqlEventSource = (EventSource)typeof(SqlConnection).Assembly.GetType("System.Data.SqlEventSource")?.GetField("Log", BindingFlags.Static | BindingFlags.NonPublic)?.GetValue(null);
```

Comment:
if we make a EventSource writer (i.e FakeSql : EventSource) which mimics the SqlClient event, we could unit test, right? (similar to unit tests firing DiagnosticSource events?)

Reply:
@cijothomas I'll investigate this docker thing a bit. If it's quick to get up and running that would be great. Otherwise, I'll add some "faked" tests.

Reply:
Okay.
cc: @TimothyMothra as he'd be looking at adding E2E test pipeline soon.

Reply:
I tried to get integration tests up and running on docker but ran into a problem. This is .NET Framework specific code, so we need a Windows container. That's not supported on GitHub at the moment, from what I can tell ([ref](https://docs.github.com/en/actions/creating-actions/about-actions)). In the end, I did two things:
1) Added integration tests that will skip unless an environment variable is set.
2) Added fake event source tests. That gives us the same level of testing as the other Sql stuff.

I did get pretty far into getting the container up and running. What I would like to do is complete this PR, and then I can try to get Sql integration tests up and running for our .NET Core stuff? I think that will work because there are SQL Linux images and of course .NET Core SDK Linux images.